src/main/java/seedu/duke/command/RemoveCommand.java

```diff
@@ -0,0 +1,94 @@
+        ArgumentType.OPTION,
+        ArgumentType.VALUE
+    };
+    protected static final String COMMAND_REMOVE = "remove";
+
+    private String recordNumberStr;
+    private int recordNumberInt;
+
+    /**
+     * Constructor to validate the format for remove command.
+     * @param arguments parsed input containing options and arguments.
+     * @param recordList is the recordList.
+     * @throws CommandException contains the error messages when a incorrect format is detected.
+     */
+    public RemoveCommand(ArrayList<String> arguments, RecordList recordList) throws CommandException {
+        validateOptions(arguments, COMMAND_REMOVE, VALID_OPTIONS, VALID_OPTIONS);
+        recordNumberStr = getIndexInString(arguments);
+        recordNumberInt = getIndexInInteger(arguments, recordList);
+        validateArguments(arguments, ARGUMENT_TYPE_ORDER, COMMAND_REMOVE);
+    }
+
+    /**
+     * Get the index field in String.
+     * @param arguments parsed input containing options and arguments.
+     * @return a String containing the index of the record.
+     * @throws CommandException contains the error messages when a incorrect format is detected.
+     */
+    private String getIndexInString(ArrayList<String> arguments) throws CommandException {
+        if (!hasOption(arguments, OPTION_INDEX)) {
+            throw new CommandException("missing option: -i", COMMAND_REMOVE);
+
+        }
+        return getOptionValue(arguments, COMMAND_REMOVE, OPTION_INDEX);
+    }
+
+    /**
+     * Get the index field in Integer.
+     * @param arguments parsed input containing options and arguments.
+     * @param recordList is the recordList.
+     * @return a Integer containing the index of the record.
+     * @throws CommandException contains the error messages when a incorrect format is detected.
+     */
+    private int getIndexInInteger(ArrayList<String> arguments, RecordList recordList) throws CommandException {
+        try {
+            return validateIndex(getOptionValue(arguments, COMMAND_REMOVE, OPTION_INDEX), recordList);
+        } catch (NumberFormatException e) {
+            throw new CommandException("Index \"" + recordNumberStr + "\" is not an integer!", COMMAND_REMOVE);
+        } catch (IndexOutOfBoundsException e) {
+            throw new CommandException("Index \"" + recordNumberStr + "\" is out of bounds!", COMMAND_REMOVE);
+        }
+    }
+
+    /**
+     * Executes the delete function.
+     * Prints a message containing the record that will be removed.
+     * Removes the record at the specified index.
+     *
+     * @param recordList is the recordList.
+     * @param ui      is the Ui object that interacts with the user.
+     * @param storage is the Storage object that reads and writes to the save file.
+     */
+    @Override
+    public void execute(RecordList recordList, Ui ui, Storage storage, CreditScoreMap creditScoreMap) {
+        Record currentRecord = recordList.getRecordAt(recordNumberInt);
```

Comment:
![image](https://user-images.githubusercontent.com/61925875/113077995-d5d59180-9204-11eb-8978-93ad35429bc5.png)

Should there currentRecord:Record returned after the call to getRecordAt()?
Is it intentional that the parameters (recordList, creditScoreMap) in the saveData() call are omitted? It might be helpful to see these parameters.

Reply:
Thanks for pointing this out! Will rectify the errors.
We are still in the process of standardizing our diagrams.